hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ReplicationConfig.java

```diff
@@ -63,7 +64,14 @@ static ReplicationConfig fromTypeAndFactor(
   }
 
   static ReplicationConfig getDefault(ConfigurationSource config) {
-    return new RatisReplicationConfig(HddsProtos.ReplicationFactor.THREE);
+    String replication = config.get(OzoneConfigKeys.OZONE_REPLICATION);
+    String replType = config.get(OzoneConfigKeys.OZONE_REPLICATION_TYPE);
+    ReplicationConfig replicationConfig = null;
+    if (replication != null && replType != null) {
+      replicationConfig = ReplicationConfig
+          .fromTypeAndString(ReplicationType.valueOf(replType), replication);
+    }
```

Comment:
Not sure we get the null there. ozone-defaults xml will always contains the default values right?
Why I added null checks were to handle a test issues(TestOzoneClient) which uses InMemoryConfiguration
But in practice, we should not get null values there. do you think we have some possibility?

Reply:
I'm not sure.  Since we already have the default values defined as Java constants, using them just like in most other cases of `Configuration.get(...)` calls would be safer.

Reply:
Thanks for the explanation.

Reply:
@adoroszlai, Thanks for taking a look. Do you have any other concerns?

Reply:
No concerns.

There is a null-check in `RpcClient.createKey`, but other than that it looks like `null` is OK.

https://github.com/apache/ozone/blob/01c6165361d32c074904f43cc60fb412c9c3a693/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L771

Reply:
Currently I have not removed replication and factor parameters from ozone-default.xml, so we not get null. As part EC branch, I am removing them from ozone-default, so we will get null there and I will remove this null check there. Once I remove them, it can become null and have to remove it. However if you feel we should remove it now, I can just remove replicationConfig from that above null check. Let me know.